src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -176,6 +257,13 @@ public override bool TryGetPropertyValue(string name, out object value)
                 // If this is a nested resource, get the value from the dictionary of nested resources.
                 object deltaNestedResource = _deltaNestedResources[name];
 
+                //If DeltaSet collection, we are handling delta collections so the value will be that itself and no need to get instance value
```

Comment:
Maybe this change should go below the `Contract.Assert` statements further below?